### DISCUSSION TITLE
SK doccs are simply awful

### DISCUSSION THREAD

**Discussion by underwater** [microsoft/semantic-kernel#8473] [General]
I am trying to move from LangChain to SK, and I was very frustrated reading the docs... They are very incomplete and unhelpful, as they seem to describe various concepts at various levels of abstraction simultaneously.   The user is left without a clear mental model of the main components and how they interact.  Other frameworks like LangChain have excellent document in this regard.

Would my below understanding be  correct:

- AI Services --> generic term to refer to LLMs.
- Plugins --> Collections of related skills grouped together
- Skills --> sequence of functions being called (either code doing an API call, or a call to an LLM with a prompt)
- Tool--> Specific action within a skill to perform a task (e.g., an HTTP request, a database query).
- Planners --> a call to an LLM that generates a PLAN 
- Agents --> able to interate through steps in a plan and invoke them

Even browsing the source code of the abstractions in the SK repo just ads to the confusion.

**Comment by markat1** [comment]
+1 Yup I totally agree  - let's hope they put some effort in making it better

**Comment by evchaki** [comment]
@underwater - we will continue to work on our docs as a evolve the SDK. Thank you for the helpful feedback above.
